sdks/typescript/client/src/components/AppFrame.tsx

```diff
@@ -0,0 +1,287 @@
+  const onErrorRef = useRef(onError);
+
+  useEffect(() => {
+    onSizeChangedRef.current = onSizeChanged;
+    onLoggingMessageRef.current = onLoggingMessage;
+    onInitializedRef.current = onInitialized;
+    onErrorRef.current = onError;
+  });
+
+  // Effect 1: Set up sandbox iframe and connect AppBridge
+  useEffect(() => {
+    const sandboxUrlString = sandbox.url.href;
+
+    // If we already have an iframe set up for this sandbox URL AND the same appBridge, skip setup
+    // This preserves the iframe state across React re-renders (including StrictMode)
+    // but ensures isolation when switching to a different app/resource (different appBridge)
+    if (
+      currentSandboxUrlRef.current === sandboxUrlString &&
+      currentAppBridgeRef.current === appBridge &&
+      iframeRef.current
+    ) {
+      return;
+    }
+
+    // Reset state when setting up a new iframe/bridge to ensure isolation
+    // between different apps/resources
+    setIframeReady(false);
+    setBridgeConnected(false);
+    setError(null);
+
+    let mounted = true;
+
+    const setup = async () => {
+      try {
+        // If switching to a different sandbox URL or appBridge, clean up the old iframe first
+        if (iframeRef.current && containerRef.current?.contains(iframeRef.current)) {
+          containerRef.current.removeChild(iframeRef.current);
+          iframeRef.current = null;
+          currentSandboxUrlRef.current = null;
+          currentAppBridgeRef.current = null;
+        }
+
+        const { iframe, onReady } = await setupSandboxProxyIframe(sandbox.url);
+
+        if (!mounted) return;
+
+        iframeRef.current = iframe;
+        currentSandboxUrlRef.current = sandboxUrlString;
+        currentAppBridgeRef.current = appBridge;
+        if (containerRef.current) {
+          containerRef.current.appendChild(iframe);
+        }
+
+        await onReady;
+
+        if (!mounted) return;
+
+        // Register size change handler
+        appBridge.onsizechange = async (params) => {
+          onSizeChangedRef.current?.(params);
+          // Also update iframe size
+          if (iframeRef.current) {
+            if (params.width !== undefined) {
+              iframeRef.current.style.width = `${params.width}px`;
+            }
+            if (params.height !== undefined) {
+              iframeRef.current.style.height = `${params.height}px`;
+            }
+          }
+        };
+
+        // Hook into initialization
+        appBridge.oninitialized = () => {
+          if (!mounted) return;
+          console.log('[AppFrame] App initialized');
+          setIframeReady(true);
+          onInitializedRef.current?.({
+            appVersion: appBridge.getAppVersion(),
+            appCapabilities: appBridge.getAppCapabilities(),
+          });
+        };
+
+        // Register logging handler
+        appBridge.onloggingmessage = (params) => {
+          onLoggingMessageRef.current?.(params);
+        };
```

Comment:
```suggestion

```
this is setting the logging callback handler twice and is overriding the one passed in the AppRenderer

making the callback not to fire.